test/apps/iodemo/io_demo.cc

```diff
@@ -948,6 +948,21 @@ class P2pDemoCommon : public UcxContext {
         validate(conn, msg, iomsg_size);
     }
 
+    static inline void check_conn_id(const iomsg_t *msg, UcxConnection *conn)
```

Comment:
IMO `check` is a bad name since it changes state (updating remote_id), maybe `validate_connection`?

Reply:
`validate_msg`

Reply:
I disagree, actually we are validating connection ids mapping here, conn_id is just 1 field of msg header, io_msg also may contains its own extra data